### PR TITLE
[GNA] Fix bug with broadcasting constant layer with fq layer

### PR DESCRIPTION
**Pull request by dmitriikhurtin**
**Details:**
- Broadcast is failed in case of const->fq->add.

**Tickets:**
56491

### CODE REVIEW

File: inference-engine/tests/functional/plugin/gna/pass_tests/broadcast_const_with_fq.cpp [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    Run();
+};
+
+std::vector<std::vector<size_t>> inputShapes1 = { {1, 1, 21, 160} };
+std::vector<std::vector<size_t>> inputShapes2 = { {1, 1, 1, 160} };
+const std::vector<size_t> level = { 65535 };
+const std::vector<InferenceEngine::Precision> netPrecisions = {InferenceEngine::Precision::FP32, InferenceEngine::Precision::FP16};
+const std::vector<std::map<std::string, std::string>> configs = {
+    {
+        {"GNA_DEVICE_MODE", "GNA_SW_FP32"},
```
Comment (elilobanova) [inline inference-engine/tests/functional/plugin/gna/pass_tests/broadcast_const_with_fq.cpp:80]: it'd be good to add GNA_SW_EXACT mode too for this test

Comment (dmitriikhurtin) [reply inference-engine/tests/functional/plugin/gna/pass_tests/broadcast_const_with_fq.cpp:80]: The problem doesn't reproduce on GNA_SW_EXACT mode.

File: inference-engine/src/gna_plugin/optimizer/gna_pass_manager.cpp [hunk 2]
```diff
@@ -1609,46 +1609,53 @@ void SubstituteScaleShiftBroadCastPass::run() {
...
-        auto currentConstBlob = constLayer->blobs.find("custom")->second;
-
-        constLayer->blobs.find("custom")->second = tileBlob(currentConstBlob, eltwiseDimsSize);
-
+        auto currentConstBlob = blobsIter->second;
+        blobsIter->second = tileBlob(currentConstBlob, eltwiseDimsSize);
         constLayer->outData.front()->setDims(nextLayer->outData.front()->getDims());
         constLayer->outData.front()->setLayout(nextLayer->outData.front()->getLayout());
+        if (prevLayer != nextLayer) {
+            prevLayer->insData.front().lock()->setDims(nextLayer->outData.front()->getDims());
```
Comment (elilobanova) [inline inference-engine/src/gna_plugin/optimizer/gna_pass_manager.cpp:1656]: constLayer->outData.front()->setDims and prevLayer->insData.front().lock()->setDims do the same thing,  so here it's done twice, I think, set of output dims for FQ is enough.
At the same time layout should be set the same way as for const layer.

Comment (dmitriikhurtin) [reply inference-engine/src/gna_plugin/optimizer/gna_pass_manager.cpp:1655]: Done

**Review by elilobanova — APPROVED** [review]

**Pull request merged**